src/CommunityToolkit/Xamarin.CommunityToolkit/ObjectModel/Extensions/ObservableRangeCollectionExtension.shared.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace Xamarin.CommunityToolkit.ObjectModel
+{
+	public static class ObservableRangeCollectionExtension
+	{
+        /// <summary>
+        /// To be used in collection initializer
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static void Add<T>(this ObservableRangeCollection<T> source, IEnumerable<T> collection)
+        {
+			_ = source ?? throw new ArgumentNullException(nameof(source));
```

Comment:
not sure if we need this check here if we want to use it just when the user constructs a new ObservableRangeCollection

Reply:
It's good practice to always do null validation on first argument of an extension method.

Added test to demonstrate how this validation is useful in our case.

Reply:
PS: If this method would be inside `ObservableRangeCollection`, this test will throw `NullReferenceException` instead of `ArgumentNullException`. But for extension method appropriate exception is `ArgumentNullException`.